src/runtime/objmodel.cpp

```diff
@@ -1705,8 +1711,37 @@ void setattrInternal(Box* obj, const std::string& attr, Box* val, SetattrRewrite
         // We don't need to to the invalidation stuff in this case.
         return;
     } else {
+        // Finally, check __setattr__
+        if (obj->cls->tp_setattr) {
+            rewrite_args = NULL;
+            REWRITE_ABORTED("");
+
+            int rtn = obj->cls->tp_setattr(obj, const_cast<char*>(attr.c_str()), val);
```

Comment:
yeah, for apis that we implement it's an easy switch.

maybe we could #ifdef the typedef for setattrfunc to be const for c++, non-const for C?